### DISCUSSION TITLE
measure flow

### DISCUSSION THREAD

**Discussion by zhoujie757** [barrust/pyprobables#105] [Q&A]
How to use this code to statistics the frequency of net flow

**Comment by barrust** [comment]
Most likely you would want to use the `count-min sketch` or something similar to count elements as you process the data. This library is designed to allow probabilistic data structures to be used to answer these types of questions. 

Something like the following could do it:

```python
from probables import CountMinSketch

cms = CountMinSketch(width=100000, depth=5)  # this can be customized
for e in dataset:  
    cms.add(e)

# from here, you can then query it after the fact to get counts of data
```

Without more information I don't know what else I can do to help.

**Comment by zhoujie757** [comment]
Thank you for your reply. However, how can I query the counts of data?


学习的乐趣
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "barrust/pyprobables"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月7日(星期二) 上午8:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [barrust/pyprobables] measure flow (Discussion #105)





 
Most likely you would want to use the count-min sketch or something similar to count elements as you process the data. This library is designed to allow probabilistic data structures to be used to answer these types of questions.
 
Something like the following could do it:
 from probables import CountMinSketch cms = CountMinSketch(width=100000, depth=5)  # this can be customized for e in dataset:       cms.add(e) # from here, you can then query it after the fact to get counts of data
 
Without more information I don't know what else I can do to help.

**Comment by barrust** [comment]
It is in the [documentation](https://pyprobables.readthedocs.io/en/latest/index.html) but in essence you can use the `check()` function to get counts back.

```python
from probables import CountMinSketch

cms = CountMinSketch(width=100000, depth=5)
cms.check("github.com")
```
